c/jesrequestsjson.c

```diff
@@ -0,0 +1,490 @@
+
+  char *convertedBody = safeMalloc(bodyLength*4,"writeDatasetConvert");
+  int conversionBufferLength = bodyLength*4;
+  int translationLength;
+  int outCCSID = NATIVE_CODEPAGE;
+
+  returnCode = convertCharset(contentBody,
+                              bodyLength,
+                              CCSID_UTF_8,
+                              CHARSET_OUTPUT_USE_BUFFER,
+                              &convertedBody,
+                              conversionBufferLength,
+                              outCCSID,
+                              NULL,
+                              &translationLength,
+                              &reasonCode);
+
+  if(returnCode == 0) {  
+    int blockSize = 0x10000;
+    int maxBlockCount = (translationLength*2)/blockSize;
+    if (!maxBlockCount){
+      maxBlockCount = 0x10;
+    }
+    ShortLivedHeap *slh = makeShortLivedHeap(blockSize,maxBlockCount);
+    char errorBuffer[2048];
+    zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "Submit Job: before JSON parse dataLength=0x%x\n",bodyLength);
+    Json *json = jsonParseUnterminatedString(slh, 
+                                             convertedBody, translationLength,
+                                             errorBuffer, sizeof(errorBuffer));
+    if (json) {
+      if (jsonIsObject(json)) {
+        JsonObject *jsonObject = jsonAsObject(json);
+        char *fileValue = jsonObjectGetString(jsonObject,"file");
+        zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "fileValue...(%s)\n", fileValue);
+        if (fileValue) {
+          snprintf(fileName, sizeof(fileName), "//'%s'", fileValue);
+          zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "filename...(%s)\n", fileName);
+          respondWithSubmitDataset(response, fileName, TRUE);
+        }
+        else {
+          char *jclValue = jsonObjectGetString(jsonObject,"jcl");
+          if (jclValue) {
+            if (jclValue[0] != '/' | jclValue[1] != '/') {
+              respondWithMessage(response, HTTP_STATUS_INTERNAL_SERVER_ERROR,
+                  "BAD_REQUEST failed with RC = %d, Submit input data does not start with a slash", 8);
+            }
+            else {
+              responseWithSubmitJCLContents(response, jclValue);
+            }
+          }
+          else {
+            zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "jclValue...(%s)\n", jclValue);
+            zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "No jcl Value...(%s)\n", jclValue);
+            respondWithError(response, HTTP_STATUS_BAD_REQUEST,"POST body No jcl value");   
+          }   
+
+        }
+        /* updateDatasetWithJSON(response, jsonObject, absolutePath, etag, force); */
+      } else {
+        zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "*** INTERNAL ERROR *** message is JSON, but not an object\n");
+      }
+    }
+    else {
+      zowelog(NULL, LOG_COMP_RESTDATASET, ZOWE_LOG_DEBUG, "UPDATE DATASET: body was not JSON!\n");
```

Comment:
Why do we use this component if it's not related to data sets?

Reply:
Updated the correct component